### PR TITLE
C section

### PR DESCRIPTION
**Pull request by noamgolani**

### CODE REVIEW

File: server/middleware/validator.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+function personValidator(req, res, next) {
+  try {
+    const { name, number } = req.body;
+    if (!number) throw { status: 400, message: "Missing number" };
+    if (!name) throw { status: 400, message: "Missing name" };
+    if (
```
Comment (gabbyMill) [inline server/middleware/validator.js:6]: !+ is nice
but in my opinon !isNaN(number.split...)
is better.

File: server/app.js [hunk 2]
```diff
@@ -5,10 +5,9 @@ const app = express();
 
 morgan.token("body", (req) => JSON.stringify(req.body));
 
-const { getLength } = require("./lib/dummyDB");
-
 const personsRoute = require("./routes/personsRoute");
 const errorHandler = require("./middleware/errorHandler");
+const person = require("./models/person");
```
Comment (gabbyMill) [inline server/app.js:10]: Person with capital P to indicate a constructor/class

File: front/views/phonebook.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            ],
+            [],
+            { id: "addNew" }
+          ),
+        ]),
+        this.phonesElem,
+      ],
+      ["table"]
+    );
+    this.rootElem.innerHTML = "";
```
Comment (gabbyMill) [inline front/views/phonebook.js:59]: Try avoiding innerHtml

File: front/models/phonebook.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+export default class Model {
+  constructor(persons) {
+    this.persons = Array.from(persons);
```
Comment (gabbyMill) [inline front/models/phonebook.js:3]: Tip:
[...persons] is equivalent

File: front/models/phonebook.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    let deleted = false;
+    this.persons = this.persons.filter((person) => {
+      if (person.id === id) {
+        deleted = true;
+        return false;
+      }
+      return true;
+    });
+    if (deleted) return this.persons;
+    else return false;
```
Comment (gabbyMill) [inline front/models/phonebook.js:37]: Love them one liners eh

File: front/controllers/phonebook.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    });
+    this.view.hideLoading(loader);
+    this.view.renderAll(all);
+  }
+
+  static async getAllPersons() {
+    return await axios.get("/api/persons");
+  }
+
+  //TODO clean this
```
Comment (gabbyMill) [inline front/controllers/phonebook.js:28]: Comment

**Review by gabbyMill — COMMENTED** [review]
Very nice work.
Beware of one-liners, they may cause bugs/ errors in future when updating code.
I remember seeing somewhere that isNaN instead of + for validation is better for error reasons but I'll get back to you on that one. For sure it is better in the extreme case of '0'
(+'0') >> false
!isNaN('0') >> true

File: front/models/phonebook.js (hunk 5 above)
Comment (noamgolani) [reply front/models/phonebook.js:37]: how can you not

**Pull request merged**